File: DSA_basics/stack_queue/q18/solution.py

STACK = []

def push(a):
    STACK.append(a)

def pop():
    STACK.pop()

def peek():
    return STACK[-1]

print(STACK)
push(34)
push(2)
print(STACK)
pop()
print(STACK)
print(peek())
print(STACK)

File: DSA_basics/stack_queue/q18/solution.cpp

// 18. Implement a stack using an array (push & pop)

#include<iostream>
#include<vector>
using namespace std;

vector<int> STACK;

void push(int a){
    STACK.push_back(a);
}

void pop(){
    STACK.pop_back();
}

int peek(){
    return STACK[-1];
}

int main(){
    push(69);
    push(29);
    cout << "[";
    for(int element : STACK){
        cout << "\t" << element << " , ";
    } 
    cout << "]" << endl;
    pop();
    peek();
    cout << "[";
    for(int element : STACK){
        cout << "\t" << element << "";
    }
    cout << "]" << endl;
    return 0;
}    